### PR TITLE
DEC-1176 Add controller action for uploading media thumbnails

### PR DESCRIPTION
**Pull request by rfox2**
**Goal**: Provide a controller action that can be used to upload a media thumbnail image
**Work**: Added the method plus specs, and route

### CODE REVIEW

File: app/controllers/v1/media_controller.rb [hunk 1]
```diff
@@ -20,10 +20,27 @@ def finish_upload
...
+      @media = MediaQuery.new.public_find(media_params[:id])
+      @uploaded_image = media_params[:uploaded_image]
+      return_value = SaveMediaThumbnail.call(
+        image: media_params[:uploaded_image],
+        item: @media.items[0],
+        media: @media
+      ) if @uploaded_image.present?
+
+      status = return_value ? :ok : :unprocessable_entity
+      render json: return_value.to_json, status: status
```
Comment (jgondron) [inline app/controllers/v1/media_controller.rb]: Is this only rendering the json response from Honeypot? Or is it/should it be rendering the media object as a whole (using SerializeMedia)?

Comment (rfox2) [inline app/controllers/v1/media_controller.rb]: Done

File: config/routes.rb [hunk 2]
```diff
@@ -103,6 +103,7 @@
 
     resources :media, only: [] do
       put "finish_upload", to: "media#finish_upload"
+      put "add_thumbnail", to: "media#add_thumbnail"
```
Comment (rfox2) [inline config/routes.rb]: Done

File: app/services/serialize_media.rb [hunk 3]
```diff
@@ -4,7 +4,8 @@ def self.to_hash(media:)
     when "Image"
       SerializeImageMedia.to_hash(media: media)
     else
-      SerializeAVMedia.to_hash(media: media)
+      result = SerializeAVMedia.to_hash(media: media)
+      result
```
Comment (jgondron) [inline app/services/serialize_media.rb]: Is the result assignment needed? Seems like this is some left over debug code

Comment (rfox2) [inline app/services/serialize_media.rb]: Ugh, yes it is

**Comment by jgondron** [comment]
👍

**Comment by jonhartzler** [comment]
👍

**Pull request merged**